lib/topological_inventory/persister/workflow.rb

```diff
@@ -0,0 +1,183 @@
+require "topological_inventory/persister/logging"
+
+module TopologicalInventory
+  module Persister
+    class Workflow
+      include Logging
+
+      def initialize(persister, messaging_client, payload)
+        @persister        = persister
+        @messaging_client = messaging_client
+        # TODO(lsmola) we should be able to reconstruct the payload out of persistor? E.g. to just repeat the sweep phase
+        @payload = payload
+      end
+
+      def execute!
+        if total_parts
+          sweep_inactive_records!
+        else
+          persist_collections!
+        end
+      end
+
+      private
+
+      attr_reader :persister, :messaging_client, :payload
+      delegate :inventory_collections,
+               :manager,
+               :persist!,
+               :refresh_state_uuid,
+               :refresh_state_part_uuid,
+               :total_parts,
+               :sweep_scope,
+               :to => :persister
```

Comment:
This looks like it would be cleaner if this wasn't in a subclass but instead in the persister class.  We wouldn't need to pass the client in, we wouldn't need to delegate all this up, etc...

Reply:
hm, the persister class is big enough already? The workflow process feels separate